main.go

```diff
@@ -34,6 +34,10 @@ func must(err error) {
 	}
 }
 
+func feedWatchdog(connection net.Conn) {
+	connection.SetDeadline(time.Now().Add(time.Duration(5) * time.Minute))
```

Comment:
1) You likely want `SetReadDeadline` rather than `SetDeadline`, as we don't know if we'll be writing in time?
2) IIRC you don't need to typecast the `5` as it's a literal, which is auto-typecast for you in compilation
3) `5 * time.Minute` should probably be a constant (in `main.go`?). `IdleConnectionTimeout` or something like that?